app/controllers/notifications_controller.rb

```diff
@@ -4,20 +4,28 @@ class NotificationsController < ApplicationController
   before_action :render_home_page_unless_authenticated, only: [:index]
 
   def index
-    scope = current_user.notifications
-    scope = params[:archive].present? ? scope.archived : scope.inbox
+    notifications = current_user.notifications
+    scope = params[:archive].present? ? notifications.archived : notifications.inbox
 
     @types               = scope.distinct.group(:subject_type).count
     @statuses            = scope.distinct.group(:unread).count
     @reasons             = scope.distinct.group(:reason).count
     @unread_repositories = scope.distinct.group(:repository_full_name).count
     @starred             = scope.starred.count
 
-    scope = scope.repo(params[:repo])     if params[:repo].present?
-    scope = scope.reason(params[:reason]) if params[:reason].present?
-    scope = scope.type(params[:type])     if params[:type].present?
-    scope = scope.status(params[:status]) if params[:status].present?
-    scope = scope.starred                 if params[:starred].present?
+    scopes = {
```

Comment:
Looking at this change again, this doesn't technically work. Here's why:

The way we were doing it before was that we were adding another scope onto the query for each parameter.

This just keeps replacing the scope instead of building on top of it.